packages/cli/src/__tests__/ceramic-daemon.test.ts

```diff
@@ -384,11 +391,78 @@ describe('Ceramic interop: core <> http-client', () => {
       fetchJson(`http://localhost:${daemon.port}/api/v0/streams/${doc.id}/content`, {
         timeout: 1000,
       })
-    ).rejects.toThrow(`Http request timed out after 1000 ms`)
+    ).rejects.toThrow(/aborted/)
 
     clearTimeout(id)
   })
 
+  it('Aborts fetch through passed in AbortSignal', async () => {
+    const content1 = { test: 123 }
+    const doc = await TileDocument.create(core, content1, null, { anchor: false })
+
+    const loadStreamMock = jest.spyOn(core, 'loadStream')
+    let id = null
+    loadStreamMock.mockImplementation(() => {
+      return new Promise((resolve) => {
+        id = setTimeout(() => {
+          resolve(doc)
+        }, 2000)
+      })
+    })
+
+    const timedAbortSignal = createTimedAbortSignal(1000)
+
+    await expect(
+      fetchJson(`http://localhost:${daemon.port}/api/v0/streams/${doc.id}/content`, {
+        signal: timedAbortSignal.signal,
+      })
+    ).rejects.toThrow(/aborted/)
+
+    timedAbortSignal.clear()
+    clearTimeout(id)
+  })
+
+  it('Aborts fetch if taking too long even if given an AbortSignal that did not get aborted', async () => {
+    const content1 = { test: 123 }
+    const doc = await TileDocument.create(core, content1, null, { anchor: false })
+
+    const loadStreamMock = jest.spyOn(core, 'loadStream')
+    let id = null
+    loadStreamMock.mockImplementation(() => {
+      return new Promise((resolve) => {
+        id = setTimeout(() => {
+          resolve(doc)
+        }, 4000)
+      })
+    })
+
+    const timedAbortSignal = createTimedAbortSignal(5000)
```

Comment:
maybe don't use a timed signal but a regular signal here to make it clear that the idea is that it never gets triggered and the timeout runs instead?

Reply:
+1